docs/recipes/RemoveMobxStateTree.md

```diff
@@ -0,0 +1,139 @@
+:::
+
+3. Remove the `reactotron-mst` plugin from Reactotron's config
+
+**devtools/ReactotronConfig.ts**
+
+```diff
+--import { mst } from "reactotron-mst"
+
+...
+
+const reactotron = Reactotron.configure({
+  name: require("../../package.json").name,
+  onConnect: () => {
+    /** since this file gets hot reloaded, let's clear the past logs every time we connect */
+    Reactotron.clear()
+  },
+--}).use(
+--  mst({
+--    /** ignore some chatty `mobx-state-tree` actions  */
+--    filter: (event) => /postProcessSnapshot|@APPLY_SNAPSHOT/.test(event.name) === false,
+--  }),
+--)
+++})
+```
+
+4. Remove `observer()` wrapped components and reformat as functional React components
+
+- Do a project-wide search for `observer(` and replace each component instance with the following pattern:
+
+```diff
+--import { observer } from "mobx-react-lite"
+
+--export const WelcomeScreen: FC<WelcomeScreenProps> = observer(function WelcomeScreen(props) {
+++export const WelcomeScreen: FC<WelcomeScreenProps> = (props) => {
+    ...
+--})
+++}
+```
+
+5. Remove `useStores()` from components
+
+- Do a project-wide search for `useStores` and remove each instance.
+- If you're converting to a different state management solution, you'll need to swap the data we get from `useStores` to your new solution. Or you can swap in temporary hardcoded values to prevent crashes while you migrate. (just don't forget about it!)
+
+```diff
+--import { useStores } from "../models"
+
+const AppStack = () => {
+--  const { authenticationStore: { isAuthenticated } } = useStores()
+++  const isAuthenticated = false // TODO: TEMPORARY VALUE - replace with alternative state management solution
+```
+
+6. Update the [Ignite Generator Templates](https://docs.infinite.red/ignite-cli/concept/Generator-Templates/)!
+
+- Follow the same pattern to replace `observer()`. This will allow you to quickly generate screens and components via `npx ignite-cli generate screen NewScreen` and `npx ignite-cli generate component NewComponent` and use your updated syntax.
+- I also recommend customizing these however else you prefer!
+
+**ignite/templates/component/NAME.tsx.ejs**  
+**ignite/templates/screen/NAMEScreen.tsx.ejs**
+
+```diff
+--import { observer } from "mobx-react-lite"
+
+--export const <%= props.pascalCaseName %> = observer(function <%= props.pascalCaseName %>(props: <%= props.pascalCaseName %>Props) {
+++export const <%= props.pascalCaseName %> = (props: <%= props.pascalCaseName %>Props) => {
+    ...
+--})
+++}
+```
+
+7. Remove old Mobx-State-Tree store initialization & hydration code in `app.tsx`.
+
+- We still need to call `hideSplashScreen` in a `useEffect` so the app loads without needing to hydrate a store first.
+
+**app/app.tsx**
+
+```diff
+--import { useInitialRootStore } from "./models"
+
+--const { rehydrated } = useInitialRootStore(() => {
+--setTimeout(hideSplashScreen, 500)
+--})
+++useEffect(() => {
```

Comment:
This needs an import adjustment on line 22 with `, { useEffect }` or refactor it to `React.useEffect`

Reply:
good catch